# ==============================================================================
# FIXING COMMIT DIFF
# ==============================================================================
# File: js/src/jit/MacroAssembler.h
# Commit: 3ecda60613c5
# Full Hash: 3ecda60613c5cb7a57cc7004919c7d12c3a29268
# Author: Lars T Hansen <lhansen@mozilla.com>
# Date: 2021-10-22 03:08:27
# Description:
#   Bug 1736531 - Fix. r=jseward
#   
#   See bug for analysis.
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D129010
# ==============================================================================

diff -r 1044b04cf89b -r 3ecda60613c5 js/src/jit/MacroAssembler.h
--- a/js/src/jit/MacroAssembler.h	Thu Oct 21 15:44:59 2021 +0000
+++ b/js/src/jit/MacroAssembler.h	Thu Oct 21 15:46:28 2021 +0000
@@ -3818,6 +3818,16 @@
                                            wasm::SymbolicAddress builtin,
                                            wasm::FailureMode failureMode);
 
+  // The System ABI frequently states that the high bits of a 64-bit register
+  // that holds a 32-bit return value are unpredictable, and C++ compilers will
+  // indeed generate code that leaves garbage in the upper bits.
+  //
+  // Adjust the contents of the 64-bit register `r` to conform to our internal
+  // convention, which requires predictable high bits.  In practice, this means
+  // that the 32-bit valuewill be zero-extended or sign-extended to 64 bits as
+  // appropriate for the platform.
+  void widenInt32(Register r) DEFINED_ON(arm64, x64, mips64);
+
   // As enterFakeExitFrame(), but using register conventions appropriate for
   // wasm stubs.
   void enterFakeExitFrameForWasm(Register cxreg, Register scratch,